### PR TITLE
Converter static traces init

### LINKED ISSUE #6690: Initial wiring of static mode traces leveraging our otelcol converter functionality
Connect the new functionality in otelcolconvert into the staticconvert package for overlapping traces configuration functionality.

### PR DESCRIPTION
**Pull request by erikbaranowski**
#### Which issue(s) this PR fixes

Closes #6690

#### PR Checklist

- [ ] CHANGELOG.md updated
- [ ] Documentation added
- [x] Tests updated
- [x] Config converters updated

### CODE REVIEW

File: internal/converter/internal/otelcolconvert/otelcolconvert.go [hunk 1]
```diff
@@ -141,9 +141,11 @@ func getFactories() otelcol.Factories {
 	return facts
 }
 
-// appendConfig converts the provided OpenTelemetry config into an equivalent
+// AppendConfig converts the provided OpenTelemetry config into an equivalent
 // Flow config and appends the result to the provided file.
-func appendConfig(file *builder.File, cfg *otelcol.Config) diag.Diagnostics {
+//
+// TODO: what prefix should each generated flow label get? each trace instance will need something unique
```
Comment (erikbaranowski) [inline internal/converter/internal/otelcolconvert/otelcolconvert.go:147]: Let's do this on the next PR with a little less noise.

File: internal/converter/internal/staticconvert/internal/build/builder.go [hunk 2]
```diff
@@ -47,36 +49,40 @@ import (
...
 	"github.com/grafana/river/token/builder"
 	"github.com/prometheus/common/model"
 	prom_config "github.com/prometheus/prometheus/config"
 	"github.com/prometheus/prometheus/model/relabel"
+	otel_component "go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/otelcol"
 )
 
-type IntegrationsConfigBuilder struct {
+type ConfigBuilder struct {
```
Comment (erikbaranowski) [inline internal/converter/internal/staticconvert/internal/build/builder.go:62]: This was renamed due to the fact this builder does:

- integrations
- server
- logging
- traces

File: internal/converter/internal/staticconvert/internal/build/global_context.go [hunk 3]
```diff
@@ -6,14 +6,14 @@ import (
 )
 
 type GlobalContext struct {
-	LabelPrefix        string
-	RemoteWriteExports *remotewrite.Exports
+	IntegrationsLabelPrefix        string
```
Comment (erikbaranowski) [inline internal/converter/internal/staticconvert/internal/build/global_context.go:9]: renames in anticipation of traces adding some to the builder global context

File: internal/converter/internal/otelcolconvert/otelcolconvert.go (hunk 1 above)
Comment (rfratto) [reply internal/converter/internal/otelcolconvert/otelcolconvert.go:147]: nit: can we move this comment to staticconvert so it's closer to where it's relevant?

File: internal/converter/internal/staticconvert/internal/build/builder.go [hunk 4]
```diff
@@ -367,6 +373,45 @@ func (b *IntegrationsConfigBuilder) appendExporterV2(commonConfig *common_v2.Met
...
+	offset := 0
+	for i, r := range otelCfg.Service.Pipelines[otel_component.NewID("traces")].Receivers {
+		if r == otel_component.NewID("push_receiver") {
+			offset = 1
+			continue
+		}
+
+		spr[i-offset] = r
+	}
+	otelCfg.Service.Pipelines[otel_component.NewID("traces")].Receivers = spr
```
Comment (rfratto) [inline internal/converter/internal/staticconvert/internal/build/builder.go:412]: I find the logic with the offset a bit hard to follow and visibly prove to be correct. Have you considered creating a slice with 0 length but the appropriate capacity so you could use appends instead? 

```go 
tracesID := otel_component.NewID("traces") 
spr := make([]otel_component.ID, 0, len(otelCfg.Service.Pipelines[tracesID].Receivers) - 1) 

for _, r := range otelCfg.Service.Pipelines[tracesID].Receivers {
    if r == otel_component.NewID("push_receiver") {
        continue 
    }    

    spr = append(spr, r)  
}

otelCfg.Service.Pipelines[tracesID].Receivers = spr 
```

**Review by rfratto — APPROVED** [review]
Very excited to start to see this finally come together!

File: internal/converter/internal/staticconvert/internal/build/builder.go (hunk 4 above)
Comment (erikbaranowski) [reply internal/converter/internal/staticconvert/internal/build/builder.go:412]: ya I'll give that a go for clarity
